File: server/ValueObjects/Player.cs
namespace WorldServer.ValueObjects 
{
    public class Player : GameObject
    {
        public Player()
        {
        }

        public Player(string name)
        {
            Name = name;
        }

        public string Name { get; private set; }
        public int Px { get; set; }
        public int Py { get; set; }
    }

}

File: client/src/models/player.ts
import { WorldClient } from '../world-client';

export interface PlayerDto {
    id: string;
    name: string;
    x: number;
    y: number;
}

export class Player implements PlayerDto {
    
    id: string;
    name: string;
    x: number;
    y: number;

    constructor(private client: WorldClient, player: PlayerDto){
        Object.assign(this, player);
    }

    move(x: number, y: number){
        this.x += x;
        this.y += y;
    }
}